src/redispatch.js

```diff
@@ -86,7 +86,7 @@ const handleEnded = (player, event) => {
     }
 
   // Prefix ended due to content ending before postroll check
-  } else if (!player.ads._contentHasEnded) {
+  } else if (!player.ads._contentHasEnded && !player.ads.stitchedAds()) {
```

Comment:
Do you need it to go into the `if (player.ads.isInAdMode()) {` block above when using stitched ads? If not, could do a

```js
if (player.ads.stitchedAds()) {
  return
}
```

below `const handleEnded = (player, event) => {`

Reply:
Yeah, it does need to go into that block because `StitchedAdRoll` needs to see `adended` to call `endLinearAdMode` and then re-trigger `ended`. Otherwise, you can end in an ad state, which is probably not desirable.

Reply:
It seems strange to cancel ended so that you can retrigger it. What if you did the above, and then handle ended event by leaving ad mode?

Reply:
I thought it might be weird to see an `ended` event during ad mode. If I were writing an integration, I would expect that `ended` did not happen during ad mode.

Reply:
Ah so you want the sequence to be adended -> leave ad mode -> ended rather than ended -> leave ad mode. That makes sense.